app/src/main/java/com/duckduckgo/app/generalsettings/GeneralSettingsViewModel.kt

```diff
@@ -0,0 +1,152 @@
+ */
+
+package com.duckduckgo.app.generalsettings
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.duckduckgo.anvil.annotations.ContributesViewModel
+import com.duckduckgo.app.pixels.AppPixelName.AUTOCOMPLETE_GENERAL_SETTINGS_TOGGLED_OFF
+import com.duckduckgo.app.pixels.AppPixelName.AUTOCOMPLETE_GENERAL_SETTINGS_TOGGLED_ON
+import com.duckduckgo.app.pixels.AppPixelName.AUTOCOMPLETE_RECENT_SITES_GENERAL_SETTINGS_TOGGLED_OFF
+import com.duckduckgo.app.pixels.AppPixelName.AUTOCOMPLETE_RECENT_SITES_GENERAL_SETTINGS_TOGGLED_ON
+import com.duckduckgo.app.settings.db.SettingsDataStore
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.di.scopes.ActivityScope
+import com.duckduckgo.history.api.NavigationHistory
+import com.duckduckgo.voice.api.VoiceSearchAvailability
+import com.duckduckgo.voice.impl.VoiceSearchPixelNames.VOICE_SEARCH_GENERAL_SETTINGS_OFF
+import com.duckduckgo.voice.impl.VoiceSearchPixelNames.VOICE_SEARCH_GENERAL_SETTINGS_ON
+import com.duckduckgo.voice.store.VoiceSearchRepository
+import javax.inject.Inject
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+
+@ContributesViewModel(ActivityScope::class)
+class GeneralSettingsViewModel @Inject constructor(
+    private val settingsDataStore: SettingsDataStore,
+    private val pixel: Pixel,
+    private val history: NavigationHistory,
+    private val voiceSearchAvailability: VoiceSearchAvailability,
+    private val voiceSearchRepository: VoiceSearchRepository,
+    private val dispatcherProvider: DispatcherProvider,
+) : ViewModel() {
+
+    data class ViewState(
+        val autoCompleteSuggestionsEnabled: Boolean,
+        val autoCompleteRecentlyVisitedSitesSuggestionsUserEnabled: Boolean,
+        val storeHistoryEnabled: Boolean,
+        val showVoiceSearch: Boolean,
+        val voiceSearchEnabled: Boolean,
+    )
+
+    private val viewState = MutableStateFlow(
+        ViewState(
+            autoCompleteSuggestionsEnabled = settingsDataStore.autoCompleteSuggestionsEnabled,
+            autoCompleteRecentlyVisitedSitesSuggestionsUserEnabled = runBlocking { history.isHistoryUserEnabled() },
+            storeHistoryEnabled = history.isHistoryFeatureAvailable(),
+            showVoiceSearch = voiceSearchAvailability.isVoiceSearchSupported,
+            voiceSearchEnabled = voiceSearchAvailability.isVoiceSearchAvailable,
+        ),
+    )
+
+    fun viewState(): Flow<ViewState> = viewState.onStart {
+        viewModelScope.launch(dispatcherProvider.io()) {
+            val autoCompleteEnabled = settingsDataStore.autoCompleteSuggestionsEnabled
+            if (!autoCompleteEnabled) {
+                history.setHistoryUserEnabled(false)
```

Comment:
QQ: why do we need to do this here?

Reply:
This logic is copied from `PrivateSearchViewModel` since it needs to have the same behavior